src/Core/SettingsEnums.h

```diff
@@ -111,6 +111,23 @@ enum QueryLogElementType : int8_t
 DECLARE_SETTING_ENUM_WITH_RENAME(LogQueriesType, QueryLogElementType)
 
 
+// Make it signed for compatibility with DataTypeEnum8
```

Comment:
Maybe more generic way will be better?

Just get the name of query AST and write it into LowCardinality(String) field?

Reply:
Otherwise we have to support this enum...

Reply:
Yeah, LC is much better